File: scripts/multi-processing/multi-processing.pl
#!/usr/bin/env perl

use v5.38;

sub worker($name) {
    say "Child process $name (PID: $$) started.";
    sleep(2);
    say "Child process $name (PID: $$) finished.";
}

say "Parent process PID: $$\n";
say "Starting 5 processes.";
my @child_pids;
for my $i (1..5) {
    my $pid = fork();

    if (!defined $pid) {
        die "Failed to fork: $!";
    } elsif ($pid == 0) {
        worker($i);
        exit;
    } else {
        push @child_pids, $pid;
    }
}

waitpid($_, 0) foreach @child_pids;

File: scripts/multi-processing/multi-processing.py
#!/usr/bin/env python3

import os
import time
import multiprocessing as mp

def worker():
    pid = os.getpid()
    print("Process " + str(pid) + " started.")
    time.sleep(2)
    print("Process " + str(pid) + " finished.")

if __name__ == '__main__':
    print('Process ID: ', os.getpid())
    print('\nStarting 5 processes.')
    for i in range(5):
        mp.Process(target=worker).start()
